crypto/bio/bio_lib.c

```diff
@@ -856,21 +863,21 @@ int BIO_connect_retry(BIO *bio, int timeout)
 
     if (rv <= 0) {
         if (get_last_sys_error() == ETIMEDOUT) {
```

Comment:
I would be surprised if that does not reset the errno,
the last call `i = BIO_sock_error(b->num);`
is not an error, so should not *change* errno,
in the end I believe using errno in this way uses undefined data?

Reply:
Originally I had used `errno` but due to code review I replaced this by `get_last_sys_error()`. 
According to `e_os.h` it (usually) does not reset the errno:
```
# define get_last_sys_error()    errno
```

Reply:
Yes, of course but errno is totally volatile, and the
value is *undefined* at least when the 
BIO_CONN_S_BLOCKED_CONNECT code path is
executed.
see `SYSerr(SYS_F_CONNECT, i);`

Reply:
For instance errno is undefined when the error
is retrieved from getsockopt(s, SOL_SOCKET, SO_ERROR,...)

https://github.com/openssl/openssl/blob/082c041b4233b17b80129d4ac6b33a28014442b0/crypto/bio/bss_conn.c#L190-L193

Reply:
This issue has been solved meanwhile, by avoiding the use of `errno`, as just mentioned above.